src/oscar/apps/offer/abstract_models.py

```diff
@@ -1112,8 +1122,23 @@ def product_queryset(self):
         Product = self.included_products.model
 
         if self.includes_all_products:
+            _filter = Q(id__in=self.excluded_products.values("id"))
+            # extend filter if excluded_categories exist
+            if self.excluded_categories.exists():
+                expanded_range_categories = ExpandDownwardsCategoryQueryset(
+                    self.excluded_categories.values("id")
+                )
+                _filter |= Q(categories__in=expanded_range_categories)
+                # extend filter for parent categories, exclude parent = None
+                if (
+                    Product.objects.exclude(parent=None)
+                    .filter(parent__categories__in=expanded_range_categories)
```

Comment:
why do we check for this? can't we just always add the extra parent query to the exclude?